llvm/lib/Transforms/Yk/ShadowStack.cpp

```diff
@@ -230,6 +230,12 @@ class YkShadowStack : public ModulePass {
         }
       }
     }
+    // Align this shadowstack frame to the natural stack alignment.
+    MaybeAlign MAlign = DL.getStackAlignment();
```

Comment:
We should probably assert that we are using the sysV ABI? Are these requirements unique to X64 sysV?

Reply:
I initially hard coded 16 bytes here, but then found that you can query the datalayout for the correct alignment. So I think this is more flexible.

Reply:
oh yeah, i missed that, my bad!